### PR TITLE
override SDL_Quit and SDL_QuitSubsystem, adjust the gesture code.

### PR DESCRIPTION
**Pull request by sezero**
@icculus, @slouken: Please review this carefully.

### CODE REVIEW

File: src/sdl2_compat.c [hunk 1]
```diff
@@ -2559,37 +2559,11 @@ GestureAddTouch(const SDL_TouchID touchId)
...
-    return 0;
-}
-
 static GestureTouch *
 GestureGetTouch(const SDL_TouchID touchId)
 {
-    int i;
-    for (i = 0; i < GestureNumTouches; i++) {
+    int i = 0;
+    for (; i < GestureNumTouches; i++) {
```
Comment (slouken) [inline src/sdl2_compat.c:2566]: Why would you make this change? It's typical for loop variables to be initialized as part of the loop construct.

File: src/sdl2_compat.c [hunk 2]
```diff
@@ -2559,37 +2559,11 @@ GestureAddTouch(const SDL_TouchID touchId)
     return &GestureTouches[GestureNumTouches++];
 }
 
-static int
-GestureDelTouch(const SDL_TouchID touchId)
```
Comment (slouken) [inline src/sdl2_compat.c:2563]: When was this function called in SDL2? I thought this was dynamically removing touches?

File: src/sdl2_compat.c (hunk 1 above)
Comment (sezero) [reply src/sdl2_compat.c:2566]: I usually do it that way if it's done only once in a procedure like this one. Can revert that part if you want.

File: src/sdl2_compat.c (hunk 2 above)
Comment (sezero) [reply src/sdl2_compat.c:2563]: Was called from within `SDL_DelTouch` in SDL_touch.c. Nowhere to call in here.

File: src/sdl2_compat.c (hunk 1 above)
Comment (slouken) [reply src/sdl2_compat.c:2566]: Yes please.

Comment (slouken) [reply src/sdl2_compat.c:2566]: This looks good to me, once you've reverted this change.

Comment (sezero) [reply src/sdl2_compat.c]: Done

**Pull request merged**